### PR TITLE
 [TS migration] Migrate 'NewChatSelector' page to TypeScript

### PR DESCRIPTION
**Pull request by MahmudjonToraqulov**
### Details

 Migrated 'NewChatSelector' page to TypeScript

### Fixed Issues

$ https://github.com/Expensify/App/issues/31989
PROPOSAL: https://github.com/Expensify/App/issues/31989#issuecomment-1883149909

### Tests

1. Open website
2. On the home page, click on plus icon
3. Click on Start Chat
4. Write any username that you didn't chatted yet 
5. Click on Enter
- [x] Verify that no errors appear in the JS console

### Offline tests

1. Open website
2. On the home page, click on plus icon
3. Click on Start Chat
4. Write any username that you didn't chatted yet 
5. Click on Enter

### QA Steps

1. Open website
2. On the home page, click on plus icon
3. Click on Start Chat
4. Write any username that you didn't chatted yet 
5. Click on Enter

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/91727683/2556756d-8f92-4050-9ca2-54e2df112c2e

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/91727683/259a20f0-9f77-4c44-a808-3e599a695e72

</details>

### CODE REVIEW

File: src/components/TabSelector/TabSelector.tsx [hunk 1]
```diff
@@ -13,7 +13,7 @@ import CONST from '@src/CONST';
 import type IconAsset from '@src/types/utils/IconAsset';
 import TabSelectorItem from './TabSelectorItem';
 
-type TabSelectorProps = {
+export type TabSelectorProps = {
```
Comment (blazejkustra) [inline src/components/TabSelector/TabSelector.tsx:16]: We don't use inline exports, please add export to the bottom of this file

File: src/libs/Navigation/OnyxTabNavigator.tsx [hunk 2]
```diff
@@ -24,11 +25,14 @@ type OnyxTabNavigatorProps = OnyxTabNavigatorOnyxProps &
         /** A function triggered when a tab has been selected */
         onTabSelected?: (newIouType: string) => void;
 
+        tabBar: (props: TabSelectorProps) => React.JSX.Element
```
Comment (blazejkustra) [inline src/libs/Navigation/OnyxTabNavigator.tsx:28]:
```suggestion
        tabBar: (props: TabSelectorProps) => ReactNode;
```

File: src/libs/Navigation/OnyxTabNavigator.tsx [hunk 3]
```diff
@@ -24,11 +25,14 @@ type OnyxTabNavigatorProps = OnyxTabNavigatorOnyxProps &
...
+
         screenListeners?: ScreenListeners<NavigationState, MaterialTopTabNavigationEventMap>;
     };
+      
 
 // eslint-disable-next-line rulesdir/no-inline-named-export
-export const TopTab = createMaterialTopTabNavigator();
+export const TopTab:any = createMaterialTopTabNavigator();
```
Comment (blazejkustra) [inline src/libs/Navigation/OnyxTabNavigator.tsx:35]: Is this PR WIP?

**Review by blazejkustra — CHANGES_REQUESTED** [review]

File: src/libs/Navigation/OnyxTabNavigator.tsx [hunk 4]
```diff
@@ -42,7 +46,7 @@ function OnyxTabNavigator({id, selectedTab = '', children, onTabSelected = () =>
             backBehavior="initialRoute"
             keyboardDismissMode="none"
             screenListeners={{
-                state: (e) => {
+                state: (e:Event) => {
```
Comment (blazejkustra) [inline src/libs/Navigation/OnyxTabNavigator.tsx:49]: Why this is needed?

**Comment by blazejkustra** [comment]
I think eslint might be failing for this one.

**Comment by blazejkustra** [comment]
Please sync this branch with the newest main. Context [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1706196639925099), thanks!

File: src/components/TabSelector/TabSelector.tsx (hunk 1 above)
Comment (MahmudjonToraqulov) [reply src/components/TabSelector/TabSelector.tsx:16]: Hey blazekustra. Ok I'll fix it.

File: src/libs/Navigation/OnyxTabNavigator.tsx (hunk 2 above)
Comment (MahmudjonToraqulov) [reply src/libs/Navigation/OnyxTabNavigator.tsx:28]: I'll fix it.

File: src/libs/Navigation/OnyxTabNavigator.tsx (hunk 3 above)
Comment (MahmudjonToraqulov) [reply src/libs/Navigation/OnyxTabNavigator.tsx:35]: Ohh, Sorry. I think it's left accidently. Let me fix it.

**Comment by blazejkustra** [comment]
@MahmudjonToraqulov Bump

**Comment by MahmudjonToraqulov** [comment]
> @MahmudjonToraqulov Bump

Hey. Sorry for the delay. I'll make it soon. Thanks

**Comment by blazejkustra** [comment]
Let me know if you have any questions or problems!

**Comment by MahmudjonToraqulov** [comment]
> Let me know if you have any questions or problems!

Hey @blazejkustra. I have had some problems with my Mac but not with the project so far. I made all changes. Could u pls check it and tell me? Thanks.

**Comment by blazejkustra** [comment]
CLA Assistant is failing @MahmudjonToraqulov 
Screenshot 2024-02-12 at 17 16 55

File: src/pages/NewChatSelectorPage.tsx [hunk 5]
```diff
@@ -46,25 +32,17 @@ function NewChatSelectorPage(props) {
             >
                 <TopTab.Screen
                     name={CONST.TAB.NEW_CHAT}
-                    component={NewChatPage}
+                    component={NewChatPage as React.ComponentType}
```
Comment (blazejkustra) [inline src/pages/NewChatSelectorPage.tsx:35]:
```suggestion
                    component={NewChatPage}
```

File: src/pages/NewChatSelectorPage.tsx [hunk 6]
```diff
@@ -46,25 +32,17 @@ function NewChatSelectorPage(props) {
...
                 />
                 <TopTab.Screen
                     name={CONST.TAB.NEW_ROOM}
-                    component={WorkspaceNewRoomPage}
+                    component={WorkspaceNewRoomPage as React.ComponentType}
```
Comment (blazejkustra) [inline src/pages/NewChatSelectorPage.tsx:39]: Let's wait for this to be merged first: https://github.com/Expensify/App/pull/34742

OR add `@ts-expect-error` here as described in TS guidelines
```suggestion
                    component={WorkspaceNewRoomPage}
```

**Review by blazejkustra — APPROVED** [review]

File: src/pages/NewChatSelectorPage.tsx (hunk 6 above)
Comment (MahmudjonToraqulov) [reply src/pages/NewChatSelectorPage.tsx:39]: > Let's wait for this to be merged first: #34742
> 
> OR add `@ts-expect-error` here as described in TS guidelines

Yes, I think we should wait for it. There was an issue with "WorkspaceNewRoomPage".

**Comment by sobitneupane** [comment]
@MahmudjonToraqulov Is this ready for review? Please let me know if it is ready for review.

**Comment by blazejkustra** [comment]
@sobitneupane We decided to [wait](https://github.com/Expensify/App/pull/34859#discussion_r1487264465) till https://github.com/Expensify/App/pull/34742 is merged

**Comment by blazejkustra** [comment]
@MahmudjonToraqulov Let's continue now!

**Comment by MahmudjonToraqulov** [comment]
> @MahmudjonToraqulov Let's continue now!

Hey @blazejkustra. I updated my branch and the error for "WorkspaceNewRoomPage" is disappeared. But there is an error for "NewChatPage" after it's migrated into typescript. 

###  **Do you have any Suggestion?**


"`NewChatPage`" page:
qq1 1



The issue in "`NewChatSelectorPage`" page:
qq1 2

**Comment by blazejkustra** [comment]
CLA Assistant is still failing 🤔

**Comment by MahmudjonToraqulov** [comment]
> CLA Assistant is still failing 🤔

I'll fix CLA issue. Right now could you check new changes and give me any suggestion with an error that is related to "NewChatPage" pls?

**Comment by blazejkustra** [comment]
I recommend dealing with CLA now, this can become cumbersome if you leave it for a longer period of time (I see some of your commits are unsigned)

**Comment by blazejkustra** [comment]
Thanks for pushing the code! I think you can use `//@ts-expect-error` like in TS guidelines (search for it in the codebase you'll find a lot of similar cases)

**Comment by MahmudjonToraqulov** [comment]
> @MahmudjonToraqulov I would like to move this PR further, please have a look at my last comment.

Hey @blazejkustra. Reall sorry for the delay. I made what you asked. Right now I'm having an issue when I make commit and  do "git push origin main". Do u know what to do?

Here is the error I'm getting ->


git1 1

**Comment by blazejkustra** [comment]
Did you read the hint from your screenshot @MahmudjonToraqulov? 🤣 I think this should help

**Comment by MahmudjonToraqulov** [comment]
> Did you read the hint from your screenshot @MahmudjonToraqulov? 🤣 I think this should help

Yes, but it didn't help so far.

**Comment by blazejkustra** [comment]
Probably you corrupted your branch and it's not compatible with what is on main, I would propose to reset your branch with git reset, or remove your branch locally and git pull from main

**Comment by blazejkustra** [comment]
Still some unverified commits:
image

**Comment by blazejkustra** [comment]
You may need to sign them manually with GPG or force push all these changes as one commit

**Comment by MahmudjonToraqulov** [comment]
> You may need to sign them manually with GPG or force push all these changes as one commit

Do you know Why is CLA failing again? What should I do to solve it?

**Comment by blazejkustra** [comment]
You need to resign first commits on this PR with GPG, [here](https://superuser.com/questions/397149/can-you-gpg-sign-old-commits) are some tips. If you aren't sure, please ask on #open-source channel for guidance 😄

**Comment by sobitneupane** [comment]
@MahmudjonToraqulov If you won't be available, we will have to assign the issue to someone else.

**Comment by sobitneupane** [comment]
Bump @MahmudjonToraqulov

**Comment by MahmudjonToraqulov** [comment]
> @MahmudjonToraqulov If you won't be available, we will have to assign the issue to someone else. We're aiming to have it resolved within the next day or two.

Hey @sobitneupane. I'm available and have finished the work already.But I'm just having an issue with CLA when making commits.  Could you help me with that?

**Comment by blazejkustra** [comment]
Have you seen my [comment](https://github.com/Expensify/App/pull/34859#issuecomment-1985382073) @MahmudjonToraqulov? Did it help by any chance?

**Comment by MahmudjonToraqulov** [comment]
> Have you seen my [comment](https://github.com/Expensify/App/pull/34859#issuecomment-1985382073) @MahmudjonToraqulov? Did it help by any chance?

Hey @blazejkustra. The main problem is that I made 3 initial commits from my browser directly So these commits got me "unverified".  

If I revert all commits and make new commits again, What do u think, will it help?

**Comment by MahmudjonToraqulov** [comment]
> Have you seen my [comment](https://github.com/Expensify/App/pull/34859#issuecomment-1985382073) @MahmudjonToraqulov? Did it help by any chance?

BTW, [this](https://github.com/Expensify/App/pull/34859#issuecomment-1985382073) way didn't help.

**Comment by blazejkustra** [comment]
> If I revert all commits and make new commits again, What do u think, will it help?

You will lose commit history, but this should work!

**Comment by MahmudjonToraqulov** [comment]
> > If I revert all commits and make new commits again, What do u think, will it help?
> 
> You will lose commit history, but this should work!

Blazejkustra This is my first time making PR for Expensify. 

Should I just comment as CLA assistent asked [here](https://github.com/Expensify/App/pull/34859#issuecomment-1902594806) to sign the CLA? 

Maybe I'm getting CLA issue as I didn't CLA yet.  Can it happen?

**Comment by sobitneupane** [comment]
There are two issues:
- Unsigned commits (For this all, commits should be signed. You can find previous discussion about it on slack. [Here](https://expensify.slack.com/archives/C02NK2DQWUX/p1707210808075859) is one)
- Unsigned CLA (For this you can follow the instruction https://github.com/Expensify/App/pull/34859#issuecomment-1902594806)

**Comment by MahmudjonToraqulov** [comment]
_**I have read the CLA Document and I hereby sign the CLA**_

**Comment by MahmudjonToraqulov** [comment]
I asked on [Slack](https://expensify.slack.com/archives/C01GTK53T8Q/p1710434004857389).  I'll wait for respond till tomorrow morning if you don't mind. Thanks.

**Comment by MahmudjonToraqulov** [comment]
Hey @blazejkustra @sobitneupane! I got [reply](https://expensify.slack.com/archives/C01GTK53T8Q/p1710434812837899?thread_ts=1710434004.857389&cid=C01GTK53T8Q) on Slack to solve the issue. 
I liked second solution. 


### What is your opinion about that?

**Comment by blazejkustra** [comment]
Both solutions are fine with me, go for a safer approach - meaning Solution 2 😄

**Comment by MahmudjonToraqulov** [comment]
I'm closing this PR. I'm gonna create new one as I we discussed [here](https://github.com/Expensify/App/pull/34859#issuecomment-1999259278).

**Pull request closed without merging**